### PR TITLE
[RFR] Fix composed field quick edition from frontend

### PR DESCRIPTION
**Pull request by Kmaschta**
[Trello card](https://trello.com/c/W36trY4p/11-les-modifications-de-valeurs-faites-%C3%A0-partir-de-la-page-daccueil-stylo-ne-sont-pas-prises-en-compte-dans-le-mod%C3%A8le-export%C3%A9-en-ad)

## TODO
- [x] Fix it
- [x] Test it

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -97,6 +97,7 @@ clear-database: ## Clear the whole database
 		db.publishedCharacteristic.remove({}); \
 		db.field.remove({}); \
 		db.dataset.remove({}); \
+		db.publishedFacet.remove({}); \
```
Comment (ThieryMichel) [inline Makefile:100]: :+1:

File: src/api/controller/api/characteristic.js [hunk 2]
```diff
@@ -8,6 +8,21 @@ import { COVER_DATASET } from '../../../common/cover';
...
+    field.composedOf.isComposedOf &&
+    field.composedOf.fields.length > 0;
+
+const isValueField = field =>
+    get(field, 'transformers[0].operation') === 'VALUE';
+
+const updateFieldValue = async (ctx, field, value) => {
+    return await ctx.field.updateOneById(
+        field._id,
+        set(field, 'transformers[0].args[0].value', value),
```
Comment (ThieryMichel) [inline src/api/controller/api/characteristic.js]: Bad idea, set mutate the object.

File: src/api/controller/api/characteristic.js [hunk 3]
```diff
@@ -16,20 +31,37 @@ export const updateCharacteristics = async ctx => {
...
+                ctx,
                 field,
-                'transformers[0].args[0].value',
-                requestedNewCharacteristics[name],
+                requestedNewCharacteristics[field.name],
+            );
+        }
+
+        if (isCompositeField(field)) {
+            const composedFields = await Promise.all(
```
Comment (ThieryMichel) [inline src/api/controller/api/characteristic.js]: use new findByNames method

**Review by ThieryMichel — CHANGES_REQUESTED** [review]

**Comment by Kmaschta** [comment]
Code review applied

File: src/api/controller/api/characteristic.js [hunk 4]
```diff
@@ -17,9 +17,11 @@ const isValueField = field =>
     get(field, 'transformers[0].operation') === 'VALUE';
 
 const updateFieldValue = async (ctx, field, value) => {
+    // lodash.set mutate the object, so we need to copy the object first
+    const copiedField = { ...field };
```
Comment (ThieryMichel) [inline src/api/controller/api/characteristic.js]: you need to deepClone the object

**Review by ThieryMichel — CHANGES_REQUESTED** [review]

**Comment by Kmaschta** [comment]
Bis :ok_hand:

**Review by ThieryMichel — APPROVED** [review]

**Pull request merged**